pkg/api/validation/validation.go

```diff
@@ -628,6 +628,9 @@ func validateHostPathVolumeSource(hostPath *api.HostPathVolumeSource, fldPath *f
 	if len(hostPath.Path) == 0 {
 		allErrs = append(allErrs, field.Required(fldPath.Child("path"), ""))
 	}
+
+	allErrs = append(allErrs, validatePathNoBacksteps(hostPath.Path, fldPath.Child("path"))...)
```

Comment:
While I am in favor of this validation, we need to ensure the following scenarios are handled well:
* if you already have a pod with a path like this, can its status still be updated? Can it be deleted gracefully?